pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -158,12 +158,20 @@ func newRestoreDataProcessor(
 			InputsToDrain: []execinfra.RowSource{input},
 			TrailingMetaCallback: func() []execinfrapb.ProducerMetadata {
 				rd.ConsumerClosed()
+				meta := &execinfrapb.ProducerMetadata{}
 				if rd.agg != nil {
-					meta := bulkutil.ConstructTracingAggregatorProducerMeta(ctx,
+					meta = bulkutil.ConstructTracingAggregatorProducerMeta(ctx,
 						rd.FlowCtx.NodeID.SQLInstanceID(), rd.FlowCtx.ID, rd.agg)
-					return []execinfrapb.ProducerMetadata{*meta}
 				}
-				return nil
+				meta.BulkProcessorProgress = &execinfrapb.RemoteProducerMetadata_BulkProcessorProgress{Drained: true}
+
+				if testingKnobs, ok := flowCtx.TestingKnobs().BackupRestoreTestingKnobs.(*sql.BackupRestoreTestingKnobs); ok {
+					if testingKnobs != nil && testingKnobs.RunAfterRestoreProcDrains != nil {
+						testingKnobs.RunAfterRestoreProcDrains()
+					}
```

Comment:
🤔 Wonder if a better name might be RunBeforeRestoreProcDrains since we are holding up processor shutdown here.